src/System.Reflection.MetadataLoadContext/src/FromCoreRt/System/DefaultBinder.cs

```diff
@@ -0,0 +1,612 @@
+                realTypes[i] = types[i].UnderlyingSystemType;
+                if (!(IsImplementedByMetadataLoadContext(realTypes[i]) || realTypes[i].IsSignatureType()))
+                    throw new ArgumentException(SR.Arg_MustBeType, nameof(types));
+            }
+            types = realTypes;
+
+            // We don't automatically jump out on exact match.
+            if (match == null || match.Length == 0)
+                throw new ArgumentException(SR.Arg_EmptyArray, nameof(match));
+
+            MethodBase[] candidates = (MethodBase[])match.Clone();
+
+            // Find all the methods that can be described by the types parameter. 
+            //  Remove all of them that cannot.
+            int CurIdx = 0;
+            for (i = 0; i < candidates.Length; i++)
+            {
+                ParameterInfo[] par = candidates[i].GetParametersNoCopy();
+                if (par.Length != types.Length)
+                    continue;
+                for (j = 0; j < types.Length; j++)
+                {
+                    Type pCls = par[j].ParameterType;
+                    if (types[j].MatchesParameterTypeExactly(par[j]))
+                        continue;
+                    if (pCls == _objectType)
+                        continue;
+
+                    Type type = types[j];
+                    if (type.IsSignatureType())
+                    {
+                        if (!(candidates[i] is MethodInfo methodInfo))
+                            break;
+                        type = type.TryResolveAgainstGenericMethod(methodInfo);
+                        if (type == null)
+                            break;
+                    }
+
+                    if (pCls.IsPrimitive)
+                    {
+                        if (!(IsImplementedByMetadataLoadContext(type.UnderlyingSystemType)) ||
+                            !CanChangePrimitive(type.UnderlyingSystemType, pCls.UnderlyingSystemType))
+                            break;
+                    }
+                    else
+                    {
+                        if (!pCls.IsAssignableFrom(type))
+                            break;
+                    }
+                }
+                if (j == types.Length)
+                    candidates[CurIdx++] = candidates[i];
+            }
+            if (CurIdx == 0)
+                return null;
+            if (CurIdx == 1)
+                return candidates[0];
+
+            // Walk all of the methods looking the most specific method to invoke
+            int currentMin = 0;
+            bool ambig = false;
+            int[] paramOrder = new int[types.Length];
+            for (i = 0; i < types.Length; i++)
+                paramOrder[i] = i;
+            for (i = 1; i < CurIdx; i++)
+            {
+                int newMin = FindMostSpecificMethod(candidates[currentMin], paramOrder, null, candidates[i], paramOrder, null, types, null);
+                if (newMin == 0)
+                    ambig = true;
```

Comment:
Nit: braces

Reply:
I fixed this particular one because of the subsequent "else" having braces. However, in general the code doesn't use braces for one-liners, and tends to have minimal extra lines\whitespace.